recipes/cairo/all/conanfile.py

```diff
@@ -0,0 +1,210 @@
+            tools.replace_in_file(win32_common, '$(FREETYPE_PATH)/lib/freetype.lib',
+                                                self.deps_cpp_info['freetype'].libs[0] + '.lib')
+            with tools.vcvars(self.settings):
+                env_msvc = VisualStudioBuildEnvironment(self)
+                env_msvc.flags.append('/FS')  # C1041 if multiple CL.EXE write to the same .PDB file, please use /FS
+                with tools.environment_append(env_msvc.vars):
+                    env_build = AutoToolsBuildEnvironment(self)
+                    args=['-f', 'Makefile.win32']
+                    args.append('CFG=%s' % str(self.settings.build_type).lower())
+                    args.append('CAIRO_HAS_FC_FONT=0')
+                    args.append('ZLIB_PATH=%s' % self.deps_cpp_info['zlib'].rootpath)
+                    args.append('LIBPNG_PATH=%s' % self.deps_cpp_info['libpng'].rootpath)
+                    args.append('PIXMAN_PATH=%s' % self.deps_cpp_info['pixman'].rootpath)
+                    args.append('FREETYPE_PATH=%s' % self.deps_cpp_info['freetype'].rootpath)
+                    args.append('GOBJECT_PATH=%s' % self.deps_cpp_info['glib'].rootpath)
+                    
+                    env_build.make(args=args)
+                    env_build.make(args=['-C', os.path.join('util', 'cairo-gobject')] + args)
+
+    def _get_env_build(self):
+        if self._env_build:
+            return self._env_build
+
+        pkg_config_path = os.path.abspath('pkgconfig')
+        pkg_config_path = tools.unix_path(pkg_config_path) if self.settings.os == 'Windows' else pkg_config_path
+
+        configure_args = ['--enable-ft' if self.options.enable_ft else '--disable-ft']
+        if self.settings.os != "Windows":
+            configure_args.append('--enable-fc' if self.options.enable_fc else '--disable-fc')
+        if self.settings.os == 'Linux':
+            configure_args.append('--enable-xlib' if self.options.enable_xlib else '--disable-xlib')
+            configure_args.append('--enable-xlib_xrender' if self.options.enable_xlib_xrender else '--disable-xlib_xrender')
+            configure_args.append('--enable-xcb' if self.options.enable_xcb else '--disable-xcb')
+        configure_args.append('--enable-gobject' if self.options.enable_glib else '--disable-gobject')
+        if self.options.shared:
+            configure_args.extend(['--disable-static', '--enable-shared'])
+        else:
+            configure_args.extend(['--enable-static', '--disable-shared'])
+
+        self._env_build = AutoToolsBuildEnvironment(self)
+        if self.settings.os == 'Macos':
+            self._env_build.link_flags.extend(['-framework CoreGraphics',
+                                            '-framework CoreFoundation'])
+        if str(self.settings.compiler) in ['gcc', 'clang', 'apple-clang']:
+            self._env_build.flags.append('-Wno-enum-conversion')
+
+        self.run('PKG_CONFIG_PATH=%s NOCONFIGURE=1 ./autogen.sh' % pkg_config_path)
```

Comment:
run_environment for m4
win_bash for mingw?

```suggestion
        self.run('PKG_CONFIG_PATH=%s NOCONFIGURE=1 ./autogen.sh' % pkg_config_path, win_bash=tools.os_info.is_windows, run_environment=True)
```